app/pages/NationalInsuranceNumberPage.scala

```diff
@@ -19,10 +19,20 @@ package pages
 import config.Refined.Max99
 import play.api.libs.json.JsPath
 import models.SchemeId.Srn
+import models.UserAnswers
+
+import scala.util.Try
 
 case class NationalInsuranceNumberPage(srn: Srn, index: Max99) extends QuestionPage[Boolean] {
 
   override def path: JsPath = JsPath \ toString
 
   override def toString: String = "nationalInsuranceNumber"
+
+  override def cleanup(value: Option[Boolean], userAnswers: UserAnswers): Try[UserAnswers] =
+    value.fold(Try(userAnswers))(if (_) {
```

Comment:
If the value is `None` it means the answer hs been removed so this case should remove both answers, something like 
```
value match {
  case Some(true)  => userAnswers.remove(NoNinoPage(srn, index))
  case Some(false) => userAnswers.remove(MemberDetailsNinoPage(srn, index))
  case None             => userAnswers.remove(NoNinoPage(srn, index))
                                     .flatMap(_.remove(MemberDetailsNinoPage(srn, index))
}
```

Reply:
Also need some tests for this 😉

Reply:
Added tests here https://github.com/hmrc/pension-scheme-return-frontend/blob/98331bf2747583df660961f6aed069355e02f4a3/test/pages/NationalInsuranceNumberPageSpec.scala#L35-L50 but good point around the None 👍